app/src/main/java/org/sagebionetworks/research/app/InstructionStep.java

```diff
@@ -106,4 +106,9 @@ public final String getType() {
 
     @NonNull
     public abstract Builder toBuilder();
+
+    @Override
+    public InstructionStep copyWithIdentifier(@NonNull String identifier) {
```

Comment:
I'm ambivalent whether we need copyWithIdentifier on Android because with AutoValue copy+update is pretty trivial.

`obj.toBuilder().setIdentifier(identifier).build()`

Reply:
Not all subclasses of Step are AutoValue and I need this method to ensure that steps have all of their containing sections identifiers prefixed onto their identifier.